docs/parser_plugins.md

```diff
@@ -1 +1,176 @@
+
+- `Evaluate`: the more active counterpart of `Map`. It inserts a function (its name given as a string) between the mapping. It passes along a list of source paths -denoted just as in `Map`- as the respective positional arguments, i.e. `*args`.
+
+### Parsing Hierarchical Tree Formats
+
+### Workflows and Super-structures
+
+Now that we have covered the SCF and standard outputs, we have to plant these 
+
+## From Text to Hierarchies
+
+Contrary to XML or HDF5, text files do not come with a machine-readable structure.
+Therefore, step 2 (targeting data fields) requires an intermediate step to fall back on _key matching_.
+The source format is converted to a temporary tree format stored just in RAM by matching lines via regular expressions.
+The keys are assigned by various `text_parser.py/Quantity`s, which are themselves grouped together in a `text_parser.py/TextParser.quantities: list[Quantity]`.
+The nesting of layers is done by referring back to `TextParser` via `Quantity.sub_parser: TextParser`.
+
+The main concern is how to leverage the additional freedom of a third format.
+Our foremost advice is to consider this tree an enriched representation of the source file and thus **follow the text file structure as faithfully as possible**.
+In its simplest form, this means
+
+1. following the order in which the data appears.
+2. capturing text blocks with a single `text_parser.py/Quantity`. Then you can further process its data via a deeper layer.
+
+Let us demonstrate the second point with FHI-aims output, which is typically partioned into blocks with headers and `|`-indented lines, e.g.
+
+```txt
+  End self-consistency iteration #     1       :  max(cpu_time)    wall_clock(cpu1)$
+  | Time for this iteration                    :        1.440 s           1.454 s$
+  | Charge density update                      :        0.500 s           0.504 s$
+  | Density mixing & preconditioning           :        0.000 s           0.000 s$
+  ...
+$
+```
+
+Here we indicate repetition as `...` and print the regex end-of-line symbol (`$`) for legibility.
+This block would be captured as
+
+```python
+
+```
+
+Note that the data extracted should be indicated between parentheses `()`, i.e. be a matching group in Python regex.
+To promote semantic extraction, we provide labelled patterns for common flags, e.g. `FHIAimsOutParser.re_float`.
+You can add a match group to a pattern via `capture(quantity)`.
+
+For blocks, the matching group should lie between the header and footer.
+Most times, devs prefer to approach the matching group content in a block as generic, non-descript text to focus on the stand-out characteristics of the block itself.
+You can use the `FHIAimsOutParser.re_non_greedy` flag here.
+Be careful: it should be superceded by a final regex pattern, else the matching continues up to the end-of-file.
```

Comment:
> Be careful: it should be superceded by a final regex pattern, else the matching continues up to the end-of-file.

Be careful: it should be ~~superceded~~ superseded by a final regex pattern ~~, else~~ ; otherwise, the matching continues ~~up~~ to the end of the file.